### ISSUE TITLE
Improve error message when using predicates in across()

### ISSUE THREAD

**Issue by iangow** [tidyverse/dbplyr#583]
`across()` seems to work for `everything` but not other predicates, such as `is.numeric()`, as seen below. 

<details>

Brief description of the problem

``` r
library(dplyr, warn.conflicts = FALSE)
library(DBI)

pg <- dbConnect(RPostgres::Postgres())

starwars %>% 
  select(-films, -vehicles, -starships) %>%
  dbWriteTable(pg, "starwars", .,
               overwrite = TRUE, row.names=FALSE)

starwars_pg <- tbl(pg, "starwars")

min_max <- list(
  min = ~min(.x, na.rm = TRUE), 
  max = ~max(.x, na.rm = TRUE)
)

starwars %>% 
  summarise(across(where(is.numeric), min_max, 
                   .names = "{.fn}.{.col}"))
#> # A tibble: 1 x 6
#>   min.height max.height min.mass max.mass min.birth_year max.birth_year
#>        <int>      <int>    <dbl>    <dbl>          <dbl>          <dbl>
#> 1         66        264       15     1358              8            896

starwars_pg %>% 
  summarise(across(where(is.numeric), min_max,
                   .names = "{.fn}.{.col}"))
#> Error: Can't rename variables in this context.

min_max <- list(
  min=min,
  max=max
)

starwars_pg %>% 
  summarise(across(where(is.numeric), min_max, na.rm=TRUE,
                   .names = "{.fn}.{.col}"))
#> Error: Can't rename variables in this context.

starwars_pg %>% 
  select_if(is.numeric) %>%
  summarise(across(everything(), min_max,
                   .names = "{.fn}.{.col}"))
#> Applying predicate on the first 100 rows
#> Warning: Missing values are always removed in SQL.
#> Use `MIN(x, na.rm = TRUE)` to silence this warning
#> This warning is displayed only once per session.
#> Warning: Missing values are always removed in SQL.
#> Use `MAX(x, na.rm = TRUE)` to silence this warning
#> This warning is displayed only once per session.
#> # Source:   lazy query [?? x 6]
#> # Database: postgres [igow@localhost:5434/crsp]
#>   min.height max.height min.mass max.mass min.birth_year max.birth_year
#>        <int>      <int>    <dbl>    <dbl>          <dbl>          <dbl>
#> 1         66        264       15     1358              8            896

starwars_pg %>% 
  summarise(across(where(is.numeric), min_max))
#> Error: Can't rename variables in this context.

starwars_pg %>%
  select_if(is.numeric) %>%
  summarize(across(everything(), min_max))
#> Applying predicate on the first 100 rows
#> # Source:   lazy query [?? x 6]
#> # Database: postgres [igow@localhost:5434/crsp]
#>   height_min height_max mass_min mass_max birth_year_min birth_year_max
#>        <int>      <int>    <dbl>    <dbl>          <dbl>          <dbl>
#> 1         66        264       15     1358              8            896

starwars_pg %>%
  summarize_if(is.numeric, list(min, max), na.rm=TRUE)
#> Applying predicate on the first 100 rows
#> # Source:   lazy query [?? x 6]
#> # Database: postgres [igow@localhost:5434/crsp]
#>   height_fn1 mass_fn1 birth_year_fn1 height_fn2 mass_fn2 birth_year_fn2
#>        <int>    <dbl>          <dbl>      <int>    <dbl>          <dbl>
#> 1         66       15              8        264     1358            896

starwars_pg %>%
  summarize_if(is.numeric, min_max, na.rm=TRUE)
#> Applying predicate on the first 100 rows
#> # Source:   lazy query [?? x 6]
#> # Database: postgres [igow@localhost:5434/crsp]
#>   height_min mass_min birth_year_min height_max mass_max birth_year_max
#>        <int>    <dbl>          <dbl>      <int>    <dbl>          <dbl>
#> 1         66       15              8        264     1358            896
```

<sup>Created on 2021-01-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

</details>

**Comment by mgirlich** [comment]
predicate functions do not work in selections on database tables. This is documented in the help page of `select.tbl_lazy()` but this might be a bit hard to find. Unfortunately, the error message you get doesn't really help you in finding out about this:

``` r
library(dplyr, warn.conflicts = FALSE)
library(dbplyr, warn.conflicts = FALSE)

starwars_db <- tbl_memdb(starwars %>% select(-films, -vehicles, -starships))
starwars_db %>% 
  select(where(is.numeric))
#> Error: Query contains no columns
```

<sup>Created on 2021-01-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The error you get actually comes from a [bug](https://github.com/r-lib/tidyselect/issues/225) in the `tidyselect` package. But even when this is fixed selecting with predicate functions is not supported in `dbplyr`.

**Comment by iangow** [comment]
@mgirlich Thanks. I think the original problem I had didn't use the predicate functions, but produced the `Error: Can't rename variables in this context.` error message (I just adapted some `starwars` code from the help that *did* happen to use those functions). So I think you have correctly identified the problem as lying elsewhere.

(Regarding the error message produced by your code, I think "predicate functions are not supported with dbplyr backends" would be a big improvement.)

**Comment by hadley** [comment]
Will need to wait on https://github.com/r-lib/tidyselect/issues/226

**Comment by mgirlich** [comment]
Already fixed in the dev version of dbplyr via #1014.
